pkg/ipam-node/controllers/ippool/ippool.go

```diff
@@ -52,15 +52,19 @@ func (r *IPPoolReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 	}
 	reqLog.Info("Notification on IPPool", "name", ipPool.Name)
 	found := false
-
 	for _, alloc := range ipPool.Status.Allocations {
 		if alloc.NodeName == r.NodeName {
+			exclusions := make([]pool.ExclusionRange, 0, len(ipPool.Spec.Exclusions))
```

Comment:
Do we need to apply similar validations like we do in the CIDRPool controller? https://github.com/Mellanox/nvidia-k8s-ipam/pull/44/files#diff-81e2300ca5363063bc54cf99edda7f4cc4db78e8ac4c73eadc34205cab124eabR100

Reply:
I believe we don't need additional logic here. The node's allocator implementation will reject configurations where exclusions are outside of the node's subnet. In the case of IPPool, all nodes use the same subnet, so any exclusions that match the pool's subnet are valid for all nodes. However, with CIDRPool, each node has a unique subnet, so we need to filter the pool's exclusions and only pass valid exclusions to the node's allocator.